File: Algorithms/Sorting/Selection Sort/selection.js
function selectionSort(numbers) {
  let minIndex,
    temp,
    len = numbers.length;

  for (let i = 0; i < len; i++) {
    minIndex = i;
    for (let j = i + 1; j < len; j++) {
      if (numbers[j] < numbers[minIndex]) {
        minIndex = j;
      }
    }
    temp = numbers[i];
    numbers[i] = numbers[minIndex];
    numbers[minIndex] = temp;
  }
  return numbers;
}

let elements = [8, 2, 1, 4, 5];

console.log("Setelah", selectionSort(elements));
// Result = [1, 2, 4, 5, 8]

File: Algorithms/Sorting/Selection Sort/selection.cpp
#include <iostream>
using namespace std;

void selectionSort(int numbers[], int size){
    int minIndex, temp;

    for (int i = 0; i < size; i++) {
        minIndex = i;
        for (int j = i + 1; j < size; j++) {
            if (numbers[j] < numbers[minIndex]) {
                minIndex = j;
            }
        }
        temp = numbers[i];
        numbers[i] = numbers[minIndex];
        numbers[minIndex] = temp;
    }
}

int main(){
    int elements[5] = {8, 2, 1, 4, 5};
    int size = sizeof(elements) / sizeof(elements[0]);

    selectionSort(elements, size);

    for (int i = 0; i < size; i++) {
        cout << elements[i] << " ";
    }
    return 0;
}
